mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -206,7 +259,53 @@ def _init_settings(self):
                     break
 
         self.settings = get_local_settings(settings_read_path, self.name)
-        self._initial_settings = deepcopy(self.settings)
+
+        # copy default values set in __init__
+        for k, v in self._initial_settings.items():
+            if k not in self.settings:
+                self.settings[k] = v
+        self._initial_settings = copy(self.settings)
+
+    def _init_filesystem(self):
+        # TODO start using self.skill_id instead of self.name
+        skill_file_system = FileSystemAccess(join('skills', self.name))
+        if not isdir(skill_file_system.path):
+            shutil.move(self.file_system.path, skill_file_system.path)
+        else:
+            for f in listdir(self.file_system.path):
+                src = join(self.file_system.path, f)
+                dst = join(skill_file_system.path, f)
+                if not exists(dst):
+                    shutil.move(src, dst)
+
+        self.file_system = skill_file_system
+
+    def handle_first_run(self):
+        intro = self.get_intro_message()
+        if intro:
+            # TODO support for returning .dialog files for easy localization
+            self.speak(intro)
+
+    def _check_for_first_run(self):
+        """The very first time a skill is run, speak the intro."""
+        first_run = self.settings.get("__mycroft_skill_firstrun", True)
+        if first_run:
+            LOG.info("First run of " + self.skill_id)
+            self.settings["__mycroft_skill_firstrun"] = False
+            save_settings(self.settings_write_path,
+                          self.settings)
+            self.handle_first_run()
+
+    @property
+    def dialog_renderer(self):
```

Comment:
Docstring should be added and maybe a return type hint

Reply:
note that this is old code, just moved between other methods, nothing changed here